### ISSUE TITLE
ENTERPRISE: Objects not caching when `backtrace` is set

### ISSUE THREAD

**Issue by gaffney** [rmosolgo/graphql-ruby#4502]
_(Dramatically simplified my previously very verbose report after much debugging and getting to the root issue)_

### Problem

Setting `backtrace` in the `context` causes caching to turn off.

### Expectation

Backtrace would have no impact on tracing. If this is expected behavior, it should be documented.

### Versions

- `graphql`: 2.0.22
- `graphql-pro`: 1.24.6
- `graphql-enterprise`: 1.3.0

**Comment by gaffney** [comment]
After taking another stab at this, I have narrowed it down to... the endpoint itself (???)

Caching is enabled when POST'ing to `/admins/graphql`, but is not enabled when posting to `/graphql`.

I am a little perplexed why that would be the case, as caching is enabled on the schema itself and has no relation (as far as I can tell) to the actual controller that runs the execution.

**Comment by gaffney** [comment]
It ended up being the `backtrace: Luna.env.local?` key in the `context`.

I am guessing this is not desired behavior?

**Comment by rmosolgo** [comment]
Hi! Sorry for the trouble and thanks for the detailed report. It looks like there was a bug in how the backtrace mixin was _prepared_ for when `backtrace: true` was provided. I've worked up a fix in #4504 -- could you give it a try and make sure it fixes it for you? You can add it to your gemfile like this: 

```ruby 
gem "graphql", github: "rmosolgo/graphql-ruby", ref: "fix-backtrace-true-with-other-traces" 
```

I'll get it into the next release soon!
